crypto/x509/x_pubkey.c

```diff
@@ -209,6 +209,11 @@ X509_PUBKEY *X509_PUBKEY_dup(const X509_PUBKEY *a)
 {
     X509_PUBKEY *pubkey = NULL;
 
+    if ((pubkey = X509_PUBKEY_new()) == NULL) {
+        ERR_raise(ERR_LIB_X509, ERR_R_MALLOC_FAILURE);
+        return NULL;
+    }
+
     if (!x509_pubkey_ex_new(NULL, ASN1_ITEM_rptr(X509_PUBKEY_INTERNAL))
```

Comment:
This will still crash.  You need to pass &pubkey in as the first argument.  The pointer is only not dereferenced on error.

Reply:
I think it should be (ASN1_VALUE **)&pubkey
(like I mentioned this as another approach in the description of this PR)